src/__tests__/checkVersion.test.ts

```diff
@@ -0,0 +1,203 @@
+import {describe, test} from 'mocha'
+import {expect} from 'chai'
+import sinon from 'sinon'
+import fs from 'fs/promises'
+import * as core from '@actions/core'
+import {CheckVersion} from '../checkVersion'
+import {GetFiles} from '../getFiles'
+
+type Tag = {
+  name: string
+  commit: {
+    sha: string
+    url: string
+  }
+  zipball_url: string
+  tarball_url: string
+  node_id: string
+}
+
+describe('checkVersion', function () {
+  afterEach(() => {
+    // Restore the default sandbox here
+    sinon.restore()
+  })
+
+  test('reads files in location errors if it does not find two files with package in name', async function () {
+    const readdirStub = sinon.stub(fs, 'readdir').resolves([])
+    let error: Error | unknown = null
+    try {
+      const getFiles = new GetFiles(fs)
+      await getFiles.getFiles('some/location')
+    } catch (err: any) {
+      if (err instanceof Error) {
+        error = err
+      }
+    }
+
+    expect(error).instanceOf(Error)
+  })
+
+  test('compare versions failed stubx not called - same', async function () {
+    const setFailedStubx = sinon.stub(core, 'setFailed')
+    const readdirStub = sinon.stub(fs, 'readdir').resolves([])
+    const checkVersion = new CheckVersion(core, fs)
+
+    await checkVersion.compareVersions('1.1.1', '1.1.1')
+
+    expect(setFailedStubx.calledOnce).to.equal(false)
+  })
+
+  test('compare versions failed stubx not called - not the same ', async function () {
+    const setFailedStubx = sinon.stub(core, 'setFailed')
+
+    const checkVersion = new CheckVersion(core, fs)
+
+    await checkVersion.compareVersions('1.1.1', '2.1.1')
+
+    expect(setFailedStubx.calledOnce).to.equal(true)
+  })
+
+  test('filter through an array of tags and return sorted ones per semver rules', async function () {
+    const checkVersion = new CheckVersion(core, fs)
+
+    const dummyData = [
```

Comment:
might be better to move these large fixtures into their own file so the tests are easier to read